src/components/NewItemForm.jsx

```diff
@@ -36,14 +59,39 @@ const NewItemForm = () => {
   };
 
   const handleDialogClose = (dialogState) => {
-    setAnchorEl(null);
+    handleMenuClose();
+    clearAllErrors();
+    clearAllInputs();
     dialogState(false);
   };
 
+  const navigate = useNavigate();
   // Collections Form
   const [collectionName, setCollectionName] = useState("");
-  const onSaveCollection = () => {
-    console.log(collectionName);
+  const [collectionError, setCollectionError] = useState(clearError);
+  const onSaveCollection = async () => {
+    if (collectionName === "") {
+      setCollectionError(requiredFieldsError);
+      return;
+    } else {
+      setCollectionError(clearError);
+    }
+
+    try {
+      let response = await postCollections(collectionName);
+      setCollectionName("");
+      handleDialogClose(setCollectionsDialogOpen);
+      (prevData) => [...prevData, data1];
```

Comment:
In my text editor it says data1 is not defined. Do you see that error in your editor?

Reply:
Yes! Sorry this was a testing line that should have been removed